test/mpi/main.cc

```diff
@@ -602,7 +603,14 @@ int main(int argc, char *argv[])
     }
 
     if (has_onesided) {
-        test->set_colls(onesided_colls);
+        std::vector<ucc_coll_type_t>           os_colls(onesided_colls.size());
+        std::vector<ucc_coll_type_t>::iterator it_start;
+
+        it_start = std::set_intersection(
```

Comment:
set_intersection requires both ranges to be sorted, I'm not sure this is always the case

Reply:
You're right. I've added in sorting of the list of collectives to test.